packages/spatial/src/common/functions/OnBeforeCompilePlugin.ts

```diff
@@ -175,6 +178,17 @@ export function overrideOnBeforeCompile() {
 
     Material.prototype._onBeforeCompile = function (shader, renderer) {
       if (!this.shader) this.shader = shader
+
+      const rendererEntity = Engine.instance.viewerEntity
```

Comment:
Again here, viewer entity is the main renderer but not the only one.

If need be, we can assign our own properties to the renderer to easily know if it's webgl or webgpu